### DISCUSSION TITLE
vitest take much time when launching via VSCode Debugging

### DISCUSSION THREAD

**Discussion by hayashi-ay** [vitest-dev/vitest#2003] [Q&A]
Hi there,

It takes much time to debug vitest with VSCode Debugging for my NestJS app.
I set up launch.json according to the [docs](https://vitest.dev/guide/debugging.html).

When I set a break point, it takes much time to stop there, but it stops though I need to wait more than 10 minutes.
I noticed if I omit `app = await createApp();` from my code, it soon stops at the breakpoint.
Does anyone have any idea as to the cause?

```js
 describe('test', () => {
   beforeAll(async () => { 
    app = await createApp();  // comment out this line
    ...
   }
   ...
}
```

```js
import { Test } from '@nestjs/testing';
import type { NestFastifyApplication } from '@nestjs/platform-fastify';
...

async function createNestInstance() {
  const moduleRef = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();

  const app = moduleRef.createNestApplication<NestFastifyApplication>(
    new FastifyAdapter(),
  );
  await app.init();
  await app.getHttpAdapter().getInstance().ready();
  app.useGlobalPipes(new ValidationPipe());

  const { httpAdapter } = app.get(HttpAdapterHost);
  app.useGlobalFilters(
    new AllExceptionsFilter(httpAdapter),
    new ServiceExceptionsFilter(),
  );

  return app;
}

export type HookedApp = Awaited<ReturnType<typeof createApp>>;

export async function createApp() {
  const app: NestFastifyApplication = await createNestInstance();
  const request: supertest.SuperTest<supertest.Test> = supertest(
    app.getHttpServer(),
  );
  ...
}
```

execution summary
- with debug
```shell
Test Files  1 passed (1)
     Tests  14 passed (14)
  Start at  20:12:14
  Duration  518.51s (transform 728ms, setup 0ms, collect 516.80s, tests 705ms)
```

- without debug
```
Test Files  1 passed (1)
     Tests  14 passed (14)
  Start at  20:22:38
  Duration  1.93s (transform 460ms, setup 0ms, collect 1.12s, tests 296ms)
```

**Comment by DevathiMahesh** [comment]
> Hi there,
> 
> It takes much time to debug vitest with VSCode Debugging for my NestJS app. I set up launch.json according to the [docs](https://vitest.dev/guide/debugging.html).
> 
> When I set a break point, it takes much time to stop there, but it stops though I need to wait more than 10 minutes. I noticed if I omit `app = await createApp();` from my code, it soon stops at the breakpoint. Does anyone have any idea as to the cause?
> 
> ```js
>  describe('test', () => {
>    beforeAll(async () => { 
>     app = await createApp();  // comment out this line
>     ...
>    }
>    ...
> }
> ```
> 
> ```js
> import { Test } from '@nestjs/testing';
> import type { NestFastifyApplication } from '@nestjs/platform-fastify';
> ...
> 
> async function createNestInstance() {
>   const moduleRef = await Test.createTestingModule({
>     imports: [AppModule],
>   }).compile();
> 
>   const app = moduleRef.createNestApplication<NestFastifyApplication>(
>     new FastifyAdapter(),
>   );
>   await app.init();
>   await app.getHttpAdapter().getInstance().ready();
>   app.useGlobalPipes(new ValidationPipe());
> 
>   const { httpAdapter } = app.get(HttpAdapterHost);
>   app.useGlobalFilters(
>     new AllExceptionsFilter(httpAdapter),
>     new ServiceExceptionsFilter(),
>   );
> 
>   return app;
> }
> 
> export type HookedApp = Awaited<ReturnType<typeof createApp>>;
> 
> export async function createApp() {
>   const app: NestFastifyApplication = await createNestInstance();
>   const request: supertest.SuperTest<supertest.Test> = supertest(
>     app.getHttpServer(),
>   );
>   ...
> }
> ```
> 
> execution summary
> 
> * with debug
> 
> ```shell
> Test Files  1 passed (1)
>      Tests  14 passed (14)
>   Start at  20:12:14
>   Duration  518.51s (transform 728ms, setup 0ms, collect 516.80s, tests 705ms)
> ```
> 
> * without debug
> 
> ```
> Test Files  1 passed (1)
>      Tests  14 passed (14)
>   Start at  20:22:38
>   Duration  1.93s (transform 460ms, setup 0ms, collect 1.12s, tests 296ms)
> ```

@hayashi-ay what is with debug & without debug means?? I could see a huge difference in collect metric. can you explain a little bit more??
